packages/workbox-routing/_default.mjs

```diff
@@ -54,20 +60,65 @@ class DefaultRouter extends Router {
    */
   registerRoute(capture, handler, method = 'GET') {
     let route;
-    // TODO Should we allow Express Route?
-    // TODO If so - don't forget to add 'string' to params in jsdoc.
-    /** if (typeof capture === 'string') {
-      if (capture.length === 0) {
-        throw new WorkboxError('empty-express-string', {
-          moduleName: 'workbox-routing',
-          class: 'DefaultRouter',
-          func: 'registerRoute',
-          paramName: 'capture',
-        });
+
+    if (typeof capture === 'string') {
+      if (process.env.NODE_ENV !== 'production') {
+        if (!(capture.startsWith('/') || capture.startsWith('http'))) {
+          throw new WorkboxError('invalid-string', {
+            moduleName: 'workbox-routing',
+            className: 'DefaultRouter',
+            funcName: 'registerRoute',
+            paramName: 'capture',
+          });
+        }
+
+        // We want to only prohibit these characters in the pathname portion of
+        // the URL.
+        const valueToCheck = capture.startsWith('http') ?
+          new URL(capture).pathname :
+          capture;
+        // See https://github.com/pillarjs/path-to-regexp#parameters
+        const wildcards = '[*:?+]';
+        if (valueToCheck.match(new RegExp(`${wildcards}`))) {
+          throw new WorkboxError('invalid-wildcards', {
+            moduleName: 'workbox-routing',
+            className: 'DefaultRouter',
+            funcName: 'registerRoute',
+            paramName: 'capture',
+            wildcards,
+          });
+        }
       }
-      route = new ExpressRoute(capture, handler, method);
-    } **/
-    if (capture instanceof RegExp) {
+
+      const matchCallback = ({url}) => {
+        // If we have a path-only capture pattern...
+        if (capture.startsWith('/')) {
```

Comment:
This if statement feels error prone.

Wouldn't this have the same result:
```javascript
// If capture is full URL, new URL will do nothing, if it's '/' it'll have same origin
return (url.href === new URL(capture, location).href);
```

You could nest a debug message to call out "the pathname matches but the origin is different so if you intended a match, please use the full URL".